### PR TITLE
feat: remove lodash util functions

### PR DESCRIPTION
**Pull request by MartinKanera**
## Description

Replace utility functions that are just aliases for `Lodash` functions with our own code. Altered type definitions and added tests for all the replaced functions.

`Lodash` is still added as a dependency since it is exposed via Backbone Collection.

## Motivation and Context

 In the long run, this change will allow us to remove lodash entirely as a project dependency.

### CODE REVIEW

File: types/joint.d.ts [hunk 1]
```diff
@@ -3126,24 +3125,17 @@ export namespace util {
...
 
-    // `merge` has a weird signature
-    // typescript cannot express "any number of objects optionally followed by CustomizerFunction"
-    export function merge(destinationObject: object, sourceObject: object, customizer?: CustomizerFunction): object;
-    export function merge(destinationObject: object, sourceObject1: object, sourceObject2: object, customizer?: CustomizerFunction): object;
-    export function merge(destinationObject: object, sourceObject1: object, sourceObject2: object, sourceObject3: object, customizer?: CustomizerFunction): object;
-    export function merge(destinationObject: object, sourceObject1: object, sourceObject2: object, sourceObject3: object, sourceObject4: object, customizer?: CustomizerFunction): object;
-    // generic but less precise signature for `merge`
-    export function merge(destinationObject: object, ...sourceObjectsOptionalFinalCustomizer: SourceObjectsOptionalFinalCustomizer): object;
+    export function merge(destinationObject: object, ...sources: object[]): object;
```
Comment (kumilingus) [inline types/joint.d.ts:3138]: Why was the `customizer` parameter removed?

**Comment by kumilingus** [comment]
We also need to update the documentation. There are links to the Lodash documentation.

**Comment by alexandernst** [comment]
Ooohh, what happened to this!!? :(

**Comment by kumilingus** [comment]
No worries. We've created and merged a new PR here: https://github.com/clientIO/joint/pull/2115

**Pull request closed without merging**